packages/@headlessui-react/src/components/menu/menu.tsx

```diff
@@ -269,9 +272,22 @@ let MenuRoot = forwardRefWithAs(function Menu<TTag extends ElementType = typeof
     [menuState, close]
   )
 
-  let theirProps = props
+  let { onClose, ...theirProps } = props
```

Comment:
I'd consider exposing `onOpen` as well while you are at it

Reply:
Hmm the `onOpen` seems easier to detect because as far as I know the only way to open the menu is by clicking `Menu.Button`? So couldn't you just add a `onClick` handler to `Menu.Button`? 

Closing is harder to detect because it can be closed by clicking outside, the escape key, and choosing an option. 

At any rate if we did want the `onOpen` prop I would opt to add that in a separate PR since this one is already fairly big.